src/main/java/onlinebookstore/dto/user/UserRegistrationRequestDto.java

```diff
@@ -0,0 +1,17 @@
+package onlinebookstore.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import onlinebookstore.validation.field.match.FieldMatch;
+
+@FieldMatch
+public record UserRegistrationRequestDto(
+        Long id,
+        @NotBlank @Email String email,
```

Comment:
```suggestion
        @NotBlank @Email (see comments) String email,
```

In your case, this annotation (@Email) does not validate the email in any way. The default validation pattern in this annotation is ".*". Therefore, I would suggest considering the following options:
Set any simple regexp to validate email (@Email(regexp="you simple pattern")), create your own annotation or just delete this annotation since it doesn't do anything here